apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -1,42 +1,355 @@
+      return;
+    }
+
+    setValue('identifier', newIdentifier);
+  }, [name]);
+
+  useEffect(() => {
+    if (integrationId === undefined || providers.length === 0) {
+      return;
+    }
     const foundProvider = providers.find((provider) => provider.integrationId === integrationId);
 
-    if (foundProvider) {
-      setSelectedProvider(foundProvider);
+    if (!foundProvider) {
+      return;
     }
-  }, integrationId !== undefined && providers.length > 0);
+
+    setSelectedProvider(foundProvider);
+    reset({
+      name: foundProvider.displayName,
+      credentials: foundProvider.credentials.reduce((prev, credential) => {
+        prev[credential.key] = credential.value;
+
+        return prev;
+      }, {} as any),
+      active: foundProvider.active,
+    });
+  }, [integrationId, providers]);
+
+  if (isLoading || areEnvironmentsLoading) {
+    return (
+      <SideBarWrapper dark={isDark}>
+        <Stack
+          align="center"
+          justify="center"
+          sx={{
+            height: '100%',
+          }}
+        >
+          <Loader color={colors.error} size={32} />
+        </Stack>
+      </SideBarWrapper>
+    );
+  }
 
   if (selectedProvider === null) {
     return null;
   }
 
-  return <SideBarWrapper dark={isDark}></SideBarWrapper>;
+  return (
+    <SideBarWrapper dark={isDark}>
+      <Form
+        name={'connect-integration-form'}
+        noValidate
+        onSubmit={(e) => {
+          handleSubmit((values) => {
+            refetch();
+          });
+          e.preventDefault();
+        }}
+      >
+        <Group spacing={5}>
+          <ProviderImage providerId={selectedProvider?.providerId} />
+          <Controller
+            control={control}
+            name="name"
+            defaultValue=""
+            render={({ field }) => {
+              return (
+                <NameInput
+                  {...field}
+                  value={field.value !== undefined ? field.value : selectedProvider?.displayName}
+                  data-test-id="provider-instance-name"
+                  placeholder="Enter instance name"
+                />
+              );
+            }}
+          />
+          <ActionIcon
+            variant={'transparent'}
+            onClick={() => {
+              navigate('/integrations');
+            }}
+          >
+            <Close color={colors.B40} />
+          </ActionIcon>
+        </Group>
+        <Group mb={16} mt={16} spacing={16}>
+          <IntegrationChannel
+            name={CHANNEL_TYPE_TO_STRING[selectedProvider?.channel || ChannelTypeEnum.EMAIL]}
+            type={selectedProvider?.channel || ChannelTypeEnum.EMAIL}
+          />
+          <IntegrationEnvironmentPill
+            name={
+              environments?.find((environment) => environment._id === selectedProvider?.environmentId)?.name ||
+              'Development'
+            }
+          />
+        </Group>
+        <CenterDiv>
+          <When truthy={haveAllCredentials}>
```

Comment:
```suggestion
          <When truthy={!haveAllCredentials}>
```

Reply:
No it should when not have all credentials